lib/Interpreter/IncrementalParser.cpp

```diff
@@ -182,22 +192,32 @@ namespace cling {
     initializeVirtualFile();
   }
 
-  void
+  bool
   IncrementalParser::Initialize(llvm::SmallVectorImpl<ParseResultTransaction>&
                                 result, bool isChildInterpreter) {
     m_TransactionPool.reset(new TransactionPool(getCI()->getSema()));
+    if (!m_TransactionPool) {
+      llvm::errs() << "TransactionPool could not be created.\n";
+      return false;
+    }
+
     if (hasCodeGenerator()) {
       getCodeGenerator()->Initialize(getCI()->getASTContext());
       m_BackendPasses.reset(new BackendPasses(getCI()->getCodeGenOpts(),
                                               getCI()->getTargetOpts(),
                                               getCI()->getLangOpts()));
+      if (!m_BackendPasses) {
```

Comment:
In my opinion, assert should be reserved to the case where a contract is broken (user is passing *unallowed/unsupported* values or something *impossible* (i.e. as far as the intent of the code is concerned the code path is thought to not be reacheable).

Another consideration is that if a hard expectation is met (i.e. with the new code, m_BackendPasses can be null, the rest of the code needs to be updated to support this case by failing elegantly)

In the code here, the only case where 'if (!m_BackendPasses)' could be true is if the system is running out of memory.  So another option here is to throw an exception.